server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1778,4 +1740,124 @@ List<SearchShardIterator> getLocalShardsIterator(
+                            causeHas500Status = ExceptionsHelper.status(f.getCause()).getStatus() >= 500;
+                        }
+                        if ((f.status().getStatus() >= 500 || causeHas500Status)
+                            && ExceptionsHelper.isNodeOrShardUnavailableTypeException(f.getCause()) == false) {
+                            logger.warn("TransportSearchAction shard failure (partial results response)", f);
+                            responseCountTotalStatus = SearchResponseMetrics.ResponseCountTotalStatus.PARTIAL_FAILURE;
+                        }
+                    }
+                }
+                // increment after the delegated onResponse to ensure we don't
+                // record both a success and a failure if there is an exception
+                searchResponseMetrics.incrementResponseCount(responseCountTotalStatus);
+
+                if (collectTelemetry()) {
+                    extractCCSTelemetry(searchResponse);
+                    recordTelemetry();
+                }
+                // TODO: should this be last?
+                listener.onResponse(searchResponse);
+            } catch (Exception e) {
+                onFailure(e);
+            }
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            searchResponseMetrics.incrementResponseCount(SearchResponseMetrics.ResponseCountTotalStatus.FAILURE);
+            if (collectTelemetry()) {
+                // TODO: better failure recognition
+                Result status;
+                if (e instanceof RemoteTransportException) {
+                    status = Result.REMOTES_UNAVAILABLE;
+                } else if (task.isCancelled() || (e instanceof TaskCancelledException)) {
+                    status = Result.CANCELED;
+                } else {
+                    status = Result.UNKNOWN;
+                }
+                usageBuilder.setFailure(status);
+                // TODO: can we still get some time measurements here? Do we want to?
+                recordTelemetry();
+            }
+            listener.onFailure(e);
+        }
+
+        private void recordTelemetry() {
+            usageService.getCcsUsageHolder().updateUsage(usageBuilder.build());
+        }
+
+        /**
+         * Extract telemetry data from the search response.
+         * @param searchResponse The final response from the search.
+         */
+        private void extractCCSTelemetry(SearchResponse searchResponse) {
+            usageBuilder.took(searchResponse.getTookInMillis());
+            // TODO: what happens with the local cluster there? Are we tracking it too just like the others?
+            for (String clusterAlias : searchResponse.getClusters().getClusterAliases()) {
+                SearchResponse.Cluster cluster = searchResponse.getClusters().getCluster(clusterAlias);
+                // TODO: check if getTook is meaningful if cluster was skipped
```

Comment:
A cluster with `skipped` status will not have a took time. You can check the toXContent rendering of the SearchResponse.Clusters.

Example:

```json
        "remote1": {
          "status": "skipped",
          "indices": "*",
          "timed_out": false,
          "failures": [
            {
              "shard": -1,
              "index": null,
              "reason": {
                "type": "search_phase_execution_exception",
                "reason": "all shards failed",
                "phase": "query",
                "grouped": true,
                "failed_shards": [
                  {
                    "shard": 0,
                    "index": "remote1:blogs",
                    "node": "wACt7av_RI26NO-SIf3oqA",
...
```

Reply:
I see, then the `getTook` part needs to be moved to the `else` clause.